modular_zubbers/code/datums/weather/weather_types/acid_rain.dm

```diff
@@ -0,0 +1,154 @@
+/datum/weather/acid_rain
+	name = "acid rainstorm"
+	desc = "The planet's thunderstorms are by nature acidic, and will incinerate anyone standing beneath them without protection."
+	probability = 90
+
+	telegraph_message = span_userdanger("Thunder rumbles far above. You hear droplets drumming against the canopy. Seek shelter.")
+	telegraph_duration = 400
+	telegraph_sound = 'modular_zubbers/sound/ambience/acidrain_start.ogg'
+
+	weather_message = span_boldannounce("<i>Acidic rain pours down around you! Get inside!</i>")
+	weather_overlay = "acid_rain"
+	weather_duration_lower = 1200
+	weather_duration_upper = 2400
+	use_glow = FALSE
+	weather_sound = 'modular_zubbers/sound/ambience/acidrain_mid.ogg'
+
+	end_duration = 100
+	end_message = span_boldannounce("The downpour gradually slows to a light shower. It should be safe outside now.")
+	end_sound = 'modular_zubbers/sound/ambience/acidrain_end.ogg'
+
+	area_type = /area
+	protect_indoors = TRUE
+	target_trait = ZTRAIT_ACIDRAIN
+
+	immunity_type = TRAIT_WEATHER_IMMUNE
+
+	barometer_predictable = FALSE
+
+// since acid rain is on a station z level, add extra checks to not annoy everyone
+/datum/weather/acid_rain/can_get_alert(mob/player)
+	if(!..())
+		return FALSE
+
+	if(!is_station_level(player.z))
+		return TRUE  // bypass checks
+
+	if(isobserver(player))
+		return TRUE
+
+	if(HAS_MIND_TRAIT(player, TRAIT_DETECT_STORM))
+		return TRUE
+
+	if(istype(get_area(player), /area/mine))
+		return TRUE
+
+
+	for(var/area/acid_area in impacted_areas)
+		if(locate(acid_area) in view(player))
+			return TRUE
+
+	return FALSE
+
+///Forever acid rain, I wouldn't suggest trying this.
+/datum/weather/acid_rain/forever_storm
+	telegraph_duration = 0
+	perpetual = TRUE
+
+	probability = 0
+
+/// Copied from sand_storm.dm to get the acid rain to work properly. Surely there's a better way to do this.
+/datum/weather/acid_rain/generate_overlay_cache()
```

Comment:
This is just copied over from sandstorms/weathers in general, the two should somehow share this code, or add a non-modular change to the original and make weathers support custom icon paths